test/Datasets/datasets.jl

```diff
@@ -199,22 +205,140 @@ using DataStructures: OrderedDict
 end
 
 @testset "Saving and loading between different backends" begin
-    using NetCDF, Zarr
+    using NetCDF, Zarr, YAXArrays
     x = rand(10, 5)
     ax1 = CategoricalAxis("Ax1", string.(1:10))
     ax2 = RangeAxis("Ax2", 1:5)
-    p = tempname()
+    p = string(tempname(),".zarr")
     savecube(YAXArray([ax1, ax2], x), p, backend = :zarr)
+    @test ispath(p)
     cube1 = Cube(p)
     @test cube1.Ax1 == ax1
     @test cube1.Ax2 == ax2
     @test eltype(cube1.Ax2.values) <: Int64
     @test cube1.data == x
     p2 = string(tempname(), ".nc")
     savecube(cube1, p2, backend = :netcdf)
+    @test ispath(p2)
     cube2 = Cube(p2)
     @test cube2.Ax1 == ax1
     @test cube2.Ax2 == ax2
     @test cube2.data == x
     @test eltype(cube2.Ax2.values) <: Int64
 end
+
+@testset "Saving, loading and appending" begin
+    using YAXArrays, Zarr, NetCDF, DiskArrays
+    x,y,z = rand(10,20),rand(10),rand(10,20,5)
+    a,b,c = YAXArray.((x,y,z))
+    f = tempname()*".zarr"
+    savecube(a,f,backend=:zarr)
+    cube = Cube(f)
+    @test cube.axes == a.axes
+    @test cube.data == x
+    @test cube.chunks == a.chunks
+
+    f = tempname()*".nc";
+    savecube(a,f,backend=:netcdf)
+    cube = Cube(f)
+    @test cube.axes == a.axes
+    @test cube.data == x
+    @test cube.chunks == a.chunks
+
+    ds = Dataset(;a,b);
+    f = tempname();
+    savedataset(ds,path=f,driver=:zarr)
+    ds = open_dataset(f,driver=:zarr)
+    @test ds.a.axes == a.axes
+    @test ds.a.data == x
+    @test ds.a.chunks == a.chunks
+
+    @test ds.b.axes == b.axes
+    @test ds.b.data == y
+    @test ds.b.chunks == b.chunks
+
+    ds2 = Dataset(c = c);
+    savedataset(ds2,path=f,backend=:zarr,append=true);
+    ds = open_dataset(f, driver=:zarr)
+
+    @test ds.a.axes == a.axes
+    @test ds.a.data == x
+    @test ds.a.chunks == a.chunks
+
+    @test ds.b.axes == b.axes
+    @test ds.b.data == y
+    @test ds.b.chunks == b.chunks
+
+    @test ds.c.axes == c.axes
+    @test ds.c.data[:,:,:] == z
+    @test ds.c.chunks == c.chunks
+
+
+    d = YAXArray(zeros(Union{Missing, Int32},10,20))
+    f = tempname()
+    r = savecube(d,f,driver=:zarr,skeleton_only=true)
```

Comment:
Should we also test it for a non-missing eltype and check that everything returns typemax?